### PR TITLE
Changing Nuget Artifact greet() Method Access Role

### LINKED ISSUE #1662: Cannot use choco list on nuget repositories
# Bug Description

The default configuration has `ANONYMOUS_ROLE` set but trying to browse a nuget directory does always require authentication.

# Steps To Reproduce

1. Deploy strongbox using default configuration
2. Enable "Allow directory browsing" in `nuget-storage`
3. Try to do any `api` transaction anonomously (like `choco list`)

# Expected Behavior

Anonoymous users should be able to browse the directory as inherited by `ARTIFACTS_RESOLVE`

# Environment

* Strongbox Version: `{"strongbox":{"version":"1.0-SNAPSHOT","revision":"eccd13064b1e3bfa"}}`
* OS: `adoptopenjdk/openjdk8:jdk8u232-b09-alpine`

### PR DESCRIPTION
**Pull request by anki2189**
# Pull Request Description

This pull request closes #1662 

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactController.java [hunk 1]
```diff
@@ -406,9 +406,9 @@ private String getFeedUri(HttpServletRequest request, String storageId, String r
     @ApiOperation(value = "Used to check storage availability")
     @ApiResponses(value = { @ApiResponse(code = HttpURLConnection.HTTP_OK, message = "Storage available."),
                             @ApiResponse(code = HttpURLConnection.HTTP_UNAUTHORIZED, message = "Storage requires authorization.") })
-    @GetMapping(path = { "{storageId}/{repositoryId}", "greet" })
-    @PreAuthorize("hasAuthority('ARTIFACTS_DEPLOY')")
-    public ResponseEntity<String> greet()
+    @GetMapping(path = { "{storageId}/{repositoryId}", "checkRepositoryHealth" })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    public ResponseEntity<String> checkRepositoryHealth()
```
Comment (sbespalov) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactController.java]: @anki2189 this probably better to be named `checkRepositoryAccess`

**Comment by sbespalov** [comment]
@anki2189 can you please rebase it with latest master, this should fix the Unit Tests

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactController.java (hunk 1 above)
Comment (anki2189) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactController.java]: Changed to `checkRepositoryAccess`

**Comment by carlspring** [comment]
Thanks for these fixes, @anki2189 , and thanks for reviewing, @sbespalov ! :)

**Pull request merged**